### DISCUSSION TITLE
Can't start the program

### DISCUSSION THREAD

**Discussion by MuggerPrime** [unexpectedpanda/retool#287] [Q&A] [answered]
When I try to launch the program, it fails with an error saying "api-ms-win-crt-runtime-l1-1-0.dll" is missing.  It is in my SYSTEM32 folder.  For good measure, I even copied it into the Retool folder, but it still didn't work.  What am I doing wrong?  Win7 SP1 64-bit

**Comment by unexpectedpanda** [comment, marked as answer]
The version of Python that Retool uses doesn't support Windows 7, as it's a dead operating system no longer supported by Microsoft and considered a security risk. Retool itself is only supported on Windows 10 and up.

You might be able to work around the Python limitation with this hack for the Windows binary version of Retool, but it's at your own risk:

https://github.com/nalexandru/api-ms-win-core-path-HACK

Edit: looks like QT has also dropped support for Windows 7. QT is the library Retool uses to implement its GUI. To my understanding you need some DLLs recompiled for Windows 7 to work around that, but I don't think all of the files necessary for Retool have been done. You can find some of them here, but they won't get you all the way there:

https://forum.qt.io/topic/133002/qt-creator-6-0-1-and-qt-6-2-2-running-on-windows-7

I'd strongly recommend not using these hacks and updating to at least Windows 10 if your hardware is capable. Even [Steam is discontinuing support for Windows 7 and 8 in January 2024](https://help.steampowered.com/en/faqs/view/4784-4F2B-1321-800A).

Otherwise, if you must stay on Windows 7 for some reason, the only remaining workaround I can think of is running a recent version of Ubuntu in a virtual machine, and running Retool on that.
